driver/conv_driver.hpp

```diff
@@ -531,6 +535,9 @@ int ConvDriver<Tgpu, Tref>::GetandSetData()
 template <typename Tgpu, typename Tref>
 int ConvDriver<Tgpu, Tref>::AddCmdLineArgs()
 {
+    inflags.AddInputFlag("in_layout", 'I', "nchw", "Input Layout (Default=nchw)", "string");
+    inflags.AddInputFlag("out_layout", 'O', "nkhw", "Output Layout (Default=nkhw)", "string");
+    inflags.AddInputFlag("fil_layout", 'f', "kcyx", "Input Layout (Default=kcyx)", "string");
```

Comment:
What about 3D convolutions? "nchw" is inappropriate, for example.

Reply:
The in_layout option allows user to use any string as layout. So if it needs to be 3D convolution "nchwd", for example, user can set it. Setting it to "nchw" "nkhw" "kcyx" just so that it got defaulted to a set of coherent characters.

Once we agreed on `miopenSet4dTensorDescriptorLayout` interface that allows user to set the layout string field of a tensor, I can update the interface such that:

- Drop the `4d` in its name
- Formalize the set of characters and allowed length for layout string